dll/cpl/input/settings_page.c

```diff
@@ -619,23 +600,19 @@ OnNotifySettingsPage(HWND hwndDlg, LPARAM lParam)
 
         case PSN_APPLY:
         {
-            BOOL bRebootNeeded = IsRebootNeeded();
-
-            /* Write Input Methods list to registry */
-            if (InputList_Process() && bRebootNeeded)
+            /* If an HKL is IME, then reboot is needed */
```

Comment:
Hmmmm.... I wonder about this: Suppose there are IME layouts already installed on the system (and you've already rebooted in case you installed one). Now, you load the input.cpl and look through the input and settings pages.

The question is: would the input list have a Node whose hkl is IME (at least, the way this code is currently written)? Because if it is so, it would appear that the current code would ask the user to reboot? But if the user changed something that is unrelated to IME and that doesn't require to reboot, but clicked on "Apply", why would the code require to reboot? This doesn't make much sense to me. The code should check whether something related to IME changed... Like for example, _**new**_ IME input nodes have been added or deleted.

I see that the input nodes support flags like: `pCurrent->wFlags & INPUT_LIST_NODE_FLAG_DELETED` or `INPUT_LIST_NODE_FLAG_EDITED`, as I've seen elsewhere in the code. So it may be worth looking for that, actually, instead of just checking whether the node is an IME input.

Reply:
Fixed.